exp/api/v1beta1/webhook.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	unsetMessage     = "field is immutable, unsetting is not allowed"
+	setMessage       = "field is immutable, setting is not allowed"
+	immutableMessage = "field is immutable"
+)
+
+func ensureStringSlicesAreEqual(a []string, b []string) bool {
```

Comment:
(acknowledging that this was just copy-pasted and not changed in this PR)

It looks like this will interpret something like `["a", "a", "b"]` and `["a", "b", "b"]` as equal, which may not be expected. If some other mechanism is preventing duplicate elements or otherwise handling this appropriately then I don't think this is an issue.

Reply:
yeah I don't know what's going on here, I'm just going to fix this